velox/expression/FunctionSignature.cpp

```diff
@@ -139,6 +139,48 @@ TypeSignature parseTypeSignature(const std::string& signature) {
   return TypeSignature(baseName, std::move(nestedTypes));
 }
 
+TypePtr signatureToVeloxType(const TypeSignature& typeSignature) {
+  const auto& baseName = typeSignature.baseName();
+  if (baseName == "tinyint") {
+    return TINYINT();
+  } else if (baseName == "smallint") {
+    return SMALLINT();
+  } else if (baseName == "integer") {
```

Comment:
We may need to support Presto-specific types as well, i.e. JSON, TIMESTAMP WITH TIME ZONE.

Maybe we can use these APIs:

```
/// Returns true built-in or custom type with specified name exists.
bool hasType(const std::string& name);

/// Returns built-in or custom type with specified name and child types.
/// Returns nullptr if type with specified name doesn't exist.
TypePtr getType(
    const std::string& name,
    const std::vector<TypeParameter>& parameters);
```